12-watchtower-API.md

```diff
@@ -4,245 +4,337 @@
+- A protocol for client/server communication.
+- How to build appointments for the WatchTower, including key/locator derivation and data encryption.
 - A format for the signed receipt. 
 
 The scope of this bolt does not include: 
 
  - A payment protocol between the customer and WatchTower. 
  - WatchTower server discovery.
  
+For the rest of this document we will refer to the WatchTower as server, and the user/Ligthning node as client.
 
-## Table of Contents
+## Table of Contents 
+* [WatchTower discovery](#watchtower-discovery)
+* [WatchTower services](#watchtower-discovery)
+	* [Basic Service](#basic-service)
+	* [Quality of Service](#quality-of-service)
+* [Sending and receiving appointments](#sending-and-receiving-appointments)
+ 	* [The `appointment` message](#the-appointment-message)
+ 	* [The `appointment_accepted` message](#the-appointment_accepted-message)
+ 	* [The `appointment_rejected` message](#the-appointment_rejected-message)
+* [Quality of Service data](#quality-of-service-data)
+	* [`accountability`](#accountability)
+* [Transaction Locator and Encryption Key](#transaction-locator-and-encryption-key)
+* [Encryption Algorithms and Parameters](#encryption-algorithms-and-parameters)
+* [Payment Modes](#payment-modes)
+* [No compression of justice transaction](#no-compression-of-justice-transaction)
 
-  * [Transaction Locator and Encryption Key](#transaction-locator-and-encryption-key)
-  * [Encryption Algorithms and Parameters](#encryption-algorithms-and-parameters)
-  * [WatchTower API](#WatchTower-api)
-  * [Signed Receipt Fields](#signed-receipt-fields)
-  * [Payment Modes](#payment-modes)
-  * [Number of updates](#number-of-updates)
-  * [No compression of justice transaction](#no-compression-of-justice-transaction)
+## WatchTower discovery
+At this point we're leaving the client/server connection to be protocol agnostic. How the Lightning node finds the WatchTower or how the WatchTower announces their presence and services provided is not specified.
 
-## Connection establishment
-Connections between the client and the server can be long-lived or restarted for every single appointment.
+Therefore, we are assuming that the client and server are connected and that the client have learnt what Quality of Service (`qos`) the tower is offering. Moreover, we assume that the client has an authentication token `auth_token` to prove he's entitled to use the service if required.
 
-		+-------+                      +-------+
-		|   A   |--(1)--- wt_init ---->|   B   |
-		|       |<-(2)--- wt_init -----|       |
-		+-------+                      +-------+
+## WatchTower services
+
+### Basic Service
+The customer can hire the WatchTower to watch for breaches on the blockchain and relay a justice transaction on their behalf. The customer receives an acknowledgement when the WatchTower has accepted the job, but the hiring protocol does guarantee the transaction inclusion.
```

Comment:
it does guarantee transaction inclusion? Or it does not? Or at laest, there is no evidence produced by the protocol that the job was accepted (so no promise of quality of service in the protocol)?)

Reply:
Does not. My bad.